### PR TITLE
Environment data docs

### LINKED ISSUE #2172: Environment Data is handled in a messy way
We are fetching and storing environments in a lot of different ways.
We have some classes handling related stuff. It is quite difficult to follow.
We should try to simplify this.
Especially the order of things is hard to follow.
What happens first? Maybe make a diagram to make this clear.

- [ ] document the loading mechanisms for environment
- [ ] if possible, simplify 

Timebox to 1 day

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

I found that the latest status of this data handling is generally ok (there have been changes since the ticket was created though, so it might be thanks to that).
I've included the analysis nevertheless. If we decide to do one of the suggested improvements, I'll create a separate ticket for that and remove it from the PR, and if we don't I'll keep it there with some explanation why it wasn't implemented.

closes #2172 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] ~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Review by Stijnkool — APPROVED** [review]
Nice overview. 👍 
Yes I remember changing some things related to the environment slices in another ticket.
I introduced the `MERGE` option.
This is why the `GetEnvironments` query for the `Initializer` and `EnvSelector` can use the same slice and query.
I think what improvement 2 describes is already happening.
Because of the `MERGE strategy`, the slice is not reset to `loading`.
And if the call from the `Initializer` already returned data, that will be used.
The call will just be executed again.

We could rename some queries to make it a bit more obvious.
But the way it is now is fine for me. Nothing is incorrectly named.

**Comment by andraskvr** [comment]
Indeed the `MERGE` option is already there, and the current naming is ok for me as well. 
I added my suggestion about the slices to #2350, and we'll see when we get to that whether we implement it or not.
